src/blk/pmem/PMEMDevice.cc

```diff
@@ -54,10 +59,59 @@ int PMEMDevice::_lock()
   return 0;
 }
 
+static int pmem_check_file_type(int fd, const char *pmem_file, uint64_t *total_size)
+{
+  int rc = 0;
+  struct stat file_stat;
+  FILE *sfile;
+
+  rc = ::fstat(fd, &file_stat);
+  if (rc) {
+    return -1;
+  }
+
+  if ((file_stat.st_mode & S_IFCHR) != S_IFCHR) {
+    return -1;
+  }
+
+  char spath[PATH_MAX], npath[PATH_MAX];
+  snprintf(spath, PATH_MAX, "/sys/dev/char/%d:%d/subsystem",
+           major(file_stat.st_rdev), minor(file_stat.st_rdev));
+
+  char *real_path = realpath(spath, npath);
+  if (!real_path) {
+    return -1;
+  }
+
+  // Need to check if it is a DAX device
+  char *base_name = strrchr(real_path, '/');
+  if (!base_name || strcmp("dax", base_name + 1)) {
+    return -1;
+  }
+
+  snprintf(spath, PATH_MAX, "/sys/dev/char/%d:%d/size", major(file_stat.st_rdev), minor(file_stat.st_rdev));
+  sfile = fopen(spath, "r");
+  if (!sfile) {
+      return -1;
+  }
+
+  if (total_size != NULL) {
+    rc = fscanf(sfile, "%lu", total_size);
+    if (rc < 0) {
+      rc = -1;
+   }
+  }
+
+  fclose(sfile);
+  return 0;
+}
+
 int PMEMDevice::open(const std::string& p)
 {
   path = p;
   int r = 0;
+  int flags = 0;
```

Comment:
please define a variable closer to where it is used for better readability.

Reply:
ack